### ISSUE TITLE
[Required]/IsRequired() on properties of owned entities is ignored

### ISSUE THREAD

**Issue by salaros** [dotnet/efcore#16943]
In previous stable EF Core version (2.2.6) required properties of owned entities were marked as not null-able in the migration code. With .NET Core 3.0.0 preview 7 (6, 5 and probably previous versions too) they are marked as null-able (`nullable: true`).
My tests in [this solution](https://github.com/salaros/dotnet-ef-3.0-preview7-regressions) demonstrate it.

![image](https://user-images.githubusercontent.com/675348/62427483-642a0500-b6fc-11e9-9a47-dae75fc4e99e.png)

### Steps to reproduce
Please take a look `OwnedEntityRequired` project in [this solution](https://github.com/salaros/dotnet-ef-3.0-preview7-regressions) and `InMemoryTests.OwnedEntityRequired()` xUnit test.

```c#
public class DummyModel
{
    [Required]
    [Key, Column(Order = 0)]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required]
    public OwnedModel OwnedModel { get; set; }
}

[Owned]
public class OwnedModel
{
    [Column(nameof(RequiredField))]
    [Required]
    public string RequiredField { get; set; }
}
```

migration on EF Core 2.2.6
```c#
RequiredField = table.Column<string>(nullable: false)
```

migration on EF Core 3.0.0 preview 7
```c#
RequiredField = table.Column<string>(nullable: true)
```

### Further technical details
EF Core version: 3.0.0-preview7.19362.6
Database Provider: any (InMemory cannot be used for test, because it ignores `IsRequired()`)
Operating system: any
IDE: Visual Studio 2019 16.2

**Comment by smitpatel** [comment]
#9005

**Comment by smitpatel** [comment]
https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-3.0/breaking-changes#de

**Comment by salaros** [comment]
> #9005

@smitpatel please read both issues carefully, I'm not using table splitting.
IMO this might be a breaking change, but at the same time it's a regression (just like global filters in EF Core 3.0.0 preview 6+)

**Comment by smitpatel** [comment]
The issue has been discussed in detail in #16552

**Comment by salaros** [comment]
@ajcvickers this is pretty frustrating experience when a bot-like user with no real code on GitHub shuts down your bug reports, you spent hours to polish, by creating demonstrable samples.

I just want to understand how can we get the same functionality we had on 2.2.6 when `[Required]/IsRequired()`-marked properties are CORRECTLY created as NON-nullable fields in the migration code.

If there is not way to revert new behavior, then it's a bug/regression

**Comment by salaros** [comment]
@ajcvickers @AndriySvyryd will I be able to restore 2.2.6 behavior with [#15607](https://github.com/aspnet/EntityFrameworkCore/issues/15607) implemented and by marking owned property itself as `[Required]/IsRequired()`?

**Comment by AndriySvyryd** [comment]
@salaros Yes, #15607 will allow you to configure the columns as non-nullable.

**Comment by StefanSimion** [comment]
@salaros

**Comment by Mike-Wazowski** [comment]
@StefanSimion 
Have you found out a solution?

**Comment by salaros** [comment]
@StefanSimion @Mike-Wazowski there is not solution, unfortunately for now EF Core team is busy breaking things right before stable releases, by committing controversial stuff
https://github.com/aspnet/EntityFrameworkCore/issues/16999

**Comment by davidhenley** [comment]
I really need this functionality back.

**Comment by AndriySvyryd** [comment]
@davidhenley If you need the column to not be nullable you can modify the generated migration.

**Comment by AndriySvyryd** [comment]
Everyone that needs this functionality please vote (:+1:) for https://github.com/aspnet/EntityFrameworkCore/issues/12100

**Comment by ftathiago** [comment]
Why this issue was closed, since still not working at efcore 5.0.8?

**Comment by ajcvickers** [comment]
@ftathiago It is closed because it is a duplicate of #12100. This is fixed in 5.0. If you are seeing otherwise, then please open a _new_ issues and attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing so that we can investigate.
